src/NuGet.Core/NuGet.PackageManagement/Utility/LogUtility.cs

```diff
@@ -49,5 +51,21 @@ internal static RestoreLogMessage AsRestoreLogMessage(this IAssetsLogMessage log
                 EndColumnNumber = logMessage.EndColumnNumber
             };
         }
+
+        public static bool AreVulnerabilitiesInRestoreSummaries(IReadOnlyList<RestoreSummary> restoreSummaries)
```

Comment:
>AreVulnerabilitiesInRestoreSummaries [](http://example.com/codeflow?start=27&length=36)

I see this is only been used in a single location. Do we need the new method? 

@donnie-msft , @zivkan

Reply:
I don't see any reason not to have the method. It's always preferable to hard coding IMO.

Reply:
`RestoreSummary` is a type used only in restore, so making this a public API is of limited reuse, since anyone checking warnings from any other source means they can't use this API. (maybe they have a class that implements NuGet's `ILogger`, or MSBuild's log interface, or more likely they're reading the assets file)

Also, putting it in NuGet.PackageManagement also reduces its reuse, because it's the package which we tell customers will have the most frequent breaking changes, so we basically discourage anyone from using. It's also a package that not many customers developing apps with "NuGet functionality" will use. We don't even ship the dll in the .NET SDK, for example.

My suggestion, if the goal is to create a public API that might actually be re-used, is to use method signature `bool IsVulnerabilityCode(NuGetLogCode)`. Since `NuGetLogCode` is defined in NuGet.Common, that's probably where the utility method should go too, so that it will be available to as many customers as possible. This API sounds useful enough to consider, while also low risk enough to not need breaking changes in the future.  Otherwise my general advice is to try to avoid adding new public APIs, as we've been blocked from insertions in the past, not to mention the inconvenience other people have when they upgrade our package version and then their app no longer works.

Years ago I wrote https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis, which has a list of questions to reflect on when adding public APIs. It basically boils down to "think about who else will use the API, and how easy is it for them to use"

Reply:
I like the refactoring suggestion of `bool IsVulnerabilityCode(NuGetLogCode)`. 
To avoid the public API consideration, can it just be `internal`, for now? If customers eventually need it, we'll have a simple change to enable it.